### PR TITLE
Update pull request template for tests-in-Azure infrastructure

### PR DESCRIPTION
**Pull request by BillyONeal**
Remove out of date parts of the pull request template, and put instructions to the pull request submitter in a comment rather than in the public facing body of each request.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,29 +1,10 @@
...
-Be sure you've read README.md and understand the scope of this repo.
-
-If you're unsure about a box, leave it unchecked. A maintainer will help you.
-
-- [ ] Identifiers in product code changes are properly `_Ugly` as per
-  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
-- [ ] The STL builds successfully and all tests have passed (must be manually
-  verified by an STL maintainer before automated testing is enabled on GitHub,
-  leave this unchecked for initial submission).
-- [ ] These changes introduce no known ABI breaks (adding members, renaming
```
Comment (StephanTLavavej) [inline .github/pull_request_template.md]: Do we really want to drop the parts about `_Ugly` discipline and ABI discipline? I know I haven't written the Contribution Guidelines yet.

Comment (BillyONeal) [reply .github/pull_request_template.md]: We haven't seen this be a problem for contributors; we can always add a comment asking them to do that if it doesn't happen. We're getting a lot of PRs from repeat contributors who already understand this and for them it's just noise.

Comment (CaseyCarter) [reply .github/pull_request_template.md]: This entire template is noise for repeat contributors; that doesn't mean we shouldn't have some instructions here to help ease new contributors until we have guidelines elsewhere. I'd prefer to see these bullet points remain, but without the silly checkboxes. It's certainly no problem for repeat contributors to ignore a few more lines of text.

Comment (BillyONeal) [reply .github/pull_request_template.md]: I consider it a problem for repeat contributors to be seeing more text every time they enter a PR. I don't consider it to be a problem as long as it's in a comment; I've restored those two bullets here.

**Comment by CaseyCarter** [comment]
Thanks for your contribution!

**Pull request merged**